### ISSUE TITLE
module.parent deprecated in Node 14+

### ISSUE THREAD

**Issue by DaveyJake** [jackfranklin/gulp-load-plugins#141]
**Describe the bug**
As of Node v14.6.0, [`module.parent` is deprecated](https://nodejs.org/api/modules.html#modules_module_parent).

**To Reproduce**
1. Attempting to use the plugin in my gulpfile.babel.js produces the following error:
> TypeError: Cannot read property 'filename' of undefined at at Object.<anonymous> (/path/to/node_modules/gulp-load-plugins/index.js:170:46)

**Expected behavior**
I expect my node environment to startup like it always has.

**Desktop (please complete the following information):**
 - OS: MacOS Big Sur 11.5.2
 - Browser: Google Chrome
 - Version: 93.0.4577.63

**Temporary Fix**
I've removed the `parentDir` variable from `index.js` and placed the `path.dirname(module.parent.filename)` value inside the `config` variable on line 47.  It now looks like this:
```javascript
const config = options.config || findup('package.json', { cwd: path.dirname(module.parent.filename) });
```
I now initialize the plugin inside my `gulpfile.babel.js` like this:
```javascript
import plugins from 'gulp-load-plugins';

const $ = plugins({ config: path.resolve( __dirname, 'package.json' ) });
```

**Comment by strarsis** [comment]
+1, this is a blocker for all recent node versions! 😿

**Comment by strarsis** [comment]
````
davTypeError: Cannot read property 'filename' of undefined
````

**Comment by jameelmoses** [comment]
I’m on vacation right now, but I’ll take a look at this next week. In the meantime, happy to review a PR if you’re inclined.

**Comment by daedam** [comment]
I encountered this problem today as well, while trying to convert my gulpfile to use ESM. I am using the following in `gulpfile.mjs` to instantiate the plugin:

```js
import gulpLoadPlugins from 'gulp-load-plugins';
const plugins = gulpLoadPlugins();
```

**Comment by jameelmoses** [comment]
Happy to look into this if someone can create a repo with a reproducible issue. I'm using Node v16.14.0 right now with `gulpfile.babel.js` without issue.

**Comment by jameelmoses** [comment]
can you try this instead of the multi-line statement?

```javascript
const $ = require('gulp-load-plugins')();
```

**Comment by SCPCOEXR** [comment]
Also dealing with the issue @daedam mentioned.
Error [ERR_REQUIRE_ESM]: require() of ES Module C:\project\node_modules\gulp-imagemin\index.js from C:\project\node_modules\gulp-load-plugins\index.js not supported.
Instead change the require of C:\project\node_modules\gulp-imagemin\index.js in C:\project\node_modules\gulp-load-plugins\index.js to a dynamic import() which is available in all CommonJS modules.

Using your suggestion @jameelmoses of:
`const $ = require('gulp-load-plugins')();`

**Comment by jameelmoses** [comment]
This has been resolved in [v2.0.8](https://github.com/jackfranklin/gulp-load-plugins/releases/tag/v2.0.8)

**Comment by evgeniy-vashchuk** [comment]
For everyone for whom this option did not work:

```
config: process.env.npm_package_json
```

I resolved it by the next code:

```
import gulpLoadPlugins from 'gulp-load-plugins';
import path from 'path';
const __dirname = path.resolve();

const plugins = gulpLoadPlugins({
  config: path.resolve(__dirname, 'package.json')
});
```
